yarn-project/end-to-end/src/uniswap_trade_on_l1_from_l2.test.ts

```diff
@@ -0,0 +1,272 @@
+        selector,
+        wethL2Contract.address.toField(),
+        wethTokenPortalAddress.toField(),
+        wethAmountToBridge,
+        new Fr(3000),
+        daiL2Contract.address.toField(),
+        daiTokenPortalAddress.toField(),
+        ownerPub,
+        ownerAddress,
+        secretHash,
+        new Fr(2 ** 32 - 1),
+        ethAccount.toField(),
+        uniswapPortalAddress,
+      )
+      .send({ from: ownerAddress });
+    await withdrawTx.isMined(0, 0.1);
+    const withdrawReceipt = await withdrawTx.getReceipt();
+    expect(withdrawReceipt.status).toBe(TxStatus.MINED);
+
+    // check weth balance of owner on L2 (we first briedged `wethAmountToBridge` into L2 and now withdrew it!)
+    await expectBalanceOnL2(ownerAddress, initialBalance - transferAmount, wethL2Contract);
+
+    // 5. Consume L2 to L1 message by calling uniswapPortal.swap()
+    logger('Call swap on uniswapPortal!');
+    const daiBalanceOfPortalBefore = await daiContract.read.balanceOf([daiTokenPortalAddress.toString()]);
+    const swapArgs = [
+      wethTokenPortalAddress.toString(),
+      wethAmountToBridge,
+      3000,
+      daiTokenPortalAddress.toString(),
+      ownerAddress.toString(),
+      secretString,
+      deadline,
+      ethAccount.toString(),
+    ] as const;
+    const { result: depositDaiMessageKeyHex } = await uniswapPortal.simulate.swap(swapArgs, {
+      account: ethAccount.toString(),
+    } as any);
+    // this should also insert a message into the inbox.
+    await uniswapPortal.write.swap(swapArgs, {} as any);
+    const depositDaiMessageKey = Fr.fromString(depositDaiMessageKeyHex);
+    // weth was swapped to dai and send to portal
+    const daiBalanceOfPortalAfter = await daiContract.read.balanceOf([daiTokenPortalAddress.toString()]);
+    expect(daiBalanceOfPortalAfter).toBeGreaterThan(daiBalanceOfPortalBefore);
+    const daiAmountToBridge = daiBalanceOfPortalAfter - daiBalanceOfPortalBefore;
+
+    // Wait for the archiver to process the message
+    await delay(5000);
+    // send a transfer tx to force through rollup with the message included
+    await transferTokensOnL2(transferAmount);
+
+    // 6. claim dai on L2
+    logger('Consuming messages to mint dai on L2');
+    // Call the mint tokens function on the noir contract
+    const daiMintTx = daiL2Contract.methods
+      .mint(daiAmountToBridge, ownerPub, ownerAddress, depositDaiMessageKey, secret, ethAccount.toField())
+      .send({ from: ownerAddress });
+    await daiMintTx.isMined(0, 0.1);
+    const daiMintTxReceipt = await daiMintTx.getReceipt();
+    expect(daiMintTxReceipt.status).toBe(TxStatus.MINED);
+    await expectBalanceOnL2(ownerAddress, initialBalance + BigInt(daiAmountToBridge), daiL2Contract);
+
+    const wethBalanceAfterSwap = await getL2BalanceOf(ownerAddress, wethL2Contract);
+    const daiBalanceAfterSwap = await getL2BalanceOf(ownerAddress, daiL2Contract);
+
+    logger('WETH balance before swap: ', wethBalanceBeforeSwap.toString());
```

Comment:
We can remove the nice logs, not really needed the value are already checked from the expects, but guess they are still cool to have 🧚

Reply:
eh I like it hahah